sql/src/main/java/io/crate/expression/tablefunctions/GenerateSeries.java

```diff
@@ -71,69 +65,104 @@
+            (signature, argTypes) -> new GenerateSeries<>(
+                signature, argTypes, 1L, (x, y) -> x - y, Long::sum, (x, y) -> x / y, Long::compare)
+        );
+        module.register(
+            Signature.table(
+                NAME,
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.INTEGER.getTypeSignature(),
+                new RowType(List.of(DataTypes.INTEGER)).getTypeSignature()),
+            (signature, argTypes) -> new GenerateSeries<>(
+                signature, argTypes, 1, (x, y) -> x - y, Integer::sum, (x, y) -> x / y, Integer::compare)
+        );
+
+        // with step
+        module.register(
+            Signature.table(
+                NAME,
+                DataTypes.LONG.getTypeSignature(),
+                DataTypes.LONG.getTypeSignature(),
+                DataTypes.LONG.getTypeSignature(),
+                new RowType(List.of(DataTypes.LONG)).getTypeSignature()),
+            (signature, argTypes) -> new GenerateSeries<>(
+                signature, argTypes, 1L, (x, y) -> x - y, Long::sum, (x, y) -> x / y, Long::compare)
+        );
+        module.register(
+            Signature.table(
+                NAME,
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.INTEGER.getTypeSignature(),
+                new RowType(List.of(DataTypes.INTEGER)).getTypeSignature()),
+            (signature, argTypes) -> new GenerateSeries<>(
+                signature, argTypes, 1, (x, y) -> x - y, Integer::sum, (x, y) -> x / y, Integer::compare)
+        );
+
+        // generate_series(ts, ts, interval)
+        for (var supportedType : List.of(DataTypes.TIMESTAMP, DataTypes.TIMESTAMPZ)) {
+            module.register(
+                Signature.table(
+                    NAME,
+                    supportedType.getTypeSignature(),
+                    supportedType.getTypeSignature(),
+                    DataTypes.INTERVAL.getTypeSignature(),
+                    new RowType(List.of(supportedType)).getTypeSignature()),
+                GenerateSeriesIntervals::new
+            );
+            module.register(
+                Signature.table(
+                    NAME,
+                    supportedType.getTypeSignature(),
+                    supportedType.getTypeSignature(),
+                    new RowType(List.of(supportedType)).getTypeSignature()),
+                (signature, argTypes) -> {
                     throw new IllegalArgumentException(
-                        "generate_series(start, stop) has type `" + startType.getName() +
+                        "generate_series(start, stop) has type `" + argTypes.get(0).getName() +
                         "` for start, but requires long/int values for start and stop, " +
                         "or if used with timestamps, it requires a third argument for the step (interval)");
```

Comment:
hm shall we really catch unsupported argument types here instead of just letting it throw an unsupported function exception? I wouldn't introduce (or take further) such a special pattern, we don't do that anywhere else afaik.

Reply:
I am not sure about this case either. I did it to make this test pass https://github.com/crate/crate/blob/c455f66a0919a71d594c2d491bac0444dd1ba087/sql/src/test/java/io/crate/expression/tablefunctions/GenerateSeriesTest.java#L155

Before doing that I tried out two approaches:
1. I didn't register this function signature, then the `generate_series(ts, ts)` function call would result in `generate_series(long, long)`. I am not sure whether this behaviour is acceptable? wdyt?
2. I tried to forbid coercion on  `generate_series(long, long)` and  `generate_series(int, int)`, then the `generate_series(ts, ts)` won't be resolved of course, but also function calls like `generate_series(1, null)` won't be resolved either.

Reply:
Ah right, thx for the info.
Pitty, so we are missing a possibility to forbid coercion from a specific type. I wouldn't implement that now but will keep it in mind for maybe future adjustments.

Reply:
> Ah right, thx for the info.
> Pitty, so we are missing a possibility to forbid coercion from a specific type. I wouldn't implement that now but will keep it in mind for maybe future adjustments.

Maybe we can start a draft/meta issue for potential casting changes for 5.0. 
I think this is another case where the leniency causes trouble and we'd be better off making it more strict.

Reply:
👍 Yes, I can start with a draft for now and convert it into an issue after migrating few more functions.